client/session.go

```diff
@@ -170,8 +192,38 @@ func (s *session) setTopologyMap(topologyMap m3db.TopologyMap) error {
 	s.Lock()
 	prev := s.queues
 	s.queues = queues
-	s.setWriteOpPoolsWithLock(topologyMap)
 	s.topoMap = topologyMap
+	prevReplicas := s.replicas
+	s.replicas = topologyMap.Replicas()
+	s.quorum = topologyMap.QuorumReplicas()
+	if s.fetchBatchOpArrayArrayPool == nil ||
+		s.fetchBatchOpArrayArrayPool.Entries() != len(queues) {
+		s.fetchBatchOpArrayArrayPool = newFetchBatchOpArrayArrayPool(
+			s.opts.GetFetchBatchOpPoolSize(),
+			len(queues),
+			s.opts.GetFetchBatchOpPoolSize()/s.opts.GetFetchBatchSize())
```

Comment:
Yea makes sense. The way I'm going about it is that the total number of fetchBatchOps in the fetchBatchOps pool should be equal to number of fetchBatchOpsArrayArray * len(queues) * capacity, so we can reverse engineer what capacity should be, which is fetchBatchOpsPoolSize / (fetchBatchOpsArrayArraySize * len(queues)).